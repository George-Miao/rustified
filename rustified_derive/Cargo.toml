[package]
name        = "rustified_derive"
version     = "0.5.3"
authors     = ["George Miao <gm@miao.dev>", "Joshua Gilman <joshuagilman@gmail.com>"]
description = "A derive macro for implementing an Endpoint for rustified"
license     = "MIT"
repository  = "https://github.com/George-Miao/rustified"
edition     = "2018"

[lib]
proc-macro = true

[dependencies]
syn              = "1.0"
quote            = "1.0"
synstructure     = "0.12.5"
proc-macro2      = "1.0.28"
regex            = "1.5.4"
serde_urlencoded = "0.7.0"
