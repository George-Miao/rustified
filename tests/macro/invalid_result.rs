use rustified::endpoint::Endpoint;
use rustified_derive::Endpoint;
use serde::Serialize;

#[derive(Debug, Endpoint, Serialize)]
#[endpoint(path = "test/path", result = "DoesNotExist")]
struct Test {}

fn main() {}
