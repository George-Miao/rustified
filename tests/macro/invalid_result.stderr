error: Unknown parameter
 --> tests/macro/invalid_result.rs:6:32
  |
6 | #[endpoint(path = "test/path", result = "DoesNotExist")]
  |                                ^^^^^^

warning: unused import: `rustified::endpoint::Endpoint`
 --> tests/macro/invalid_result.rs:1:5
  |
1 | use rustified::endpoint::Endpoint;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
